src/test/java/picard/fingerprint/FingerprintCheckerTest.java

```diff
@@ -316,8 +320,45 @@ public void testQueryable(final File vcf, boolean expectedQueryable) {
         }
     }
 
+    @DataProvider
+    Object[][] deepDataProvider() {
+        return new Object[][]{{10}, {50}, {100}, {500}, {1000}, {5000}, {10000}, {50000}};
+    }
+
+    @Test(dataProvider = "deepDataProvider")
+    void testCanHandleDeepData(int counts) {
+        final HaplotypeProbabilitiesFromSequence hp1 = new HaplotypeProbabilitiesFromSequence(hb);
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele1());
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele2());
+
+        final double[] logLikelihoods = hp1.getLogLikelihoods();
+        Assert.assertTrue( MathUtil.min(logLikelihoods) < 0 );
+    }
+
+    @Test(dataProvider = "deepDataProvider")
+    void testCanHandleDeepDataHPFromContamination(int counts) throws IOException {
+        final double maf = 0.4;
+        final Snp snp = new Snp("test", "chr1", 1, (byte) 'A', (byte) 'T', maf, Collections.singletonList("dummy"));
+        final HaplotypeBlock hb = new HaplotypeBlock(maf);
+        hb.addSnp(snp);
+
+        final HaplotypeProbabilitiesFromSequence hp1 = new HaplotypeProbabilitiesFromContaminatorSequence(hb,.999);
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele1());
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele2());
+
+        final double[] logLikelihoods = hp1.getLogLikelihoods();
+        Assert.assertTrue( MathUtil.min(logLikelihoods) < 0);
+
+        final File fasta = new File(TEST_DATA_DIR, "reference.fasta");
+
+        try (final ReferenceSequenceFile ref = ReferenceSequenceFileFactory.getReferenceSequenceFile(fasta)) {
```

Comment:
Why is this also being checked here?  Isn't this redundant with just checking the logLikelihoods of the `HaplotypeProbablity`?  Or if not, why is it only in testCanHandleDeepDataHPFromContamination, but not in testCanHandleDeepData?

Reply:
I wasn't sure where the problem was....since there's a transformation between fingerprinting and the final VCF, I wanted to make sure I look at the VCF....I can refactor as you suggested above.